### ISSUE TITLE
Unable to connect to OpenVPN after fresh install  (Connection reset, restarting)

### ISSUE THREAD

**Issue by rdbox** [StreisandEffect/streisand#1579]
I used the old streisand assembly for almost 2 years, but yesterday I decided to upgrade to a new version and was very upset that a lot of errors appeared, some of them could be solved but most of the errors were not solved for me.
OpenVPN - Connection reset, restarting [0]

### Expected behavior:
Client should connected to the host using the provided OpenVPN configuration.

### Steps to Reproduce:
1. - Launched Linux droplets DigitalOcean (Ubuntu 16.04.6 LTS)
2. - Install streisand (using git clone ...)
3. - One of the openvpn configuration file
4. - Use only Windows 7 OpenVPN 
5. - Tried to connect error   Connection reset, restarting [0]

Log on clients OpenVPN GUI
```
Sat May 25 15:43:59 2019 OpenVPN 2.4.7 x86_64-w64-mingw32 [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [AEAD] built on Apr 25 2019
Sat May 25 15:43:59 2019 Windows version 6.1 (Windows 7) 64bit
Sat May 25 15:43:59 2019 library versions: OpenSSL 1.1.0j  20 Nov 2018, LZO 2.10
Sat May 25 15:43:59 2019 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:25340
Sat May 25 15:43:59 2019 Need hold release from management interface, waiting...
Sat May 25 15:44:00 2019 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:25340
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'state on'
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'log all on'
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'echo all on'
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'bytecount 5'
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'hold off'
Sat May 25 15:44:00 2019 MANAGEMENT: CMD 'hold release'
Sat May 25 15:44:00 2019 Outgoing Control Channel Encryption: Cipher 'AES-256-CTR' initialized with 256 bit key
Sat May 25 15:44:00 2019 Outgoing Control Channel Encryption: Using 256 bit message hash 'SHA256' for HMAC authentication
Sat May 25 15:44:00 2019 Incoming Control Channel Encryption: Cipher 'AES-256-CTR' initialized with 256 bit key
Sat May 25 15:44:00 2019 Incoming Control Channel Encryption: Using 256 bit message hash 'SHA256' for HMAC authentication
Sat May 25 15:44:00 2019 TCP/UDP: Preserving recently used remote address: [AF_INET]104.248.138.30:636
Sat May 25 15:44:00 2019 Socket Buffers: R=[8192->8192] S=[8192->8192]
Sat May 25 15:44:00 2019 Attempting to establish TCP connection with [AF_INET]104.248.138.30:636 [nonblock]
Sat May 25 15:44:00 2019 MANAGEMENT: >STATE:1558788240,TCP_CONNECT,,,,,,
Sat May 25 15:44:01 2019 TCP connection established with [AF_INET]104.248.138.30:636
Sat May 25 15:44:01 2019 TCP_CLIENT link local: (not bound)
Sat May 25 15:44:01 2019 TCP_CLIENT link remote: [AF_INET]104.248.138.30:636
Sat May 25 15:44:01 2019 MANAGEMENT: >STATE:1558788241,WAIT,,,,,,
Sat May 25 15:44:01 2019 MANAGEMENT: >STATE:1558788241,AUTH,,,,,,
Sat May 25 15:44:01 2019 TLS: Initial packet from [AF_INET]104.248.138.30:636, sid=986c9274 9a475882
Sat May 25 15:44:01 2019 VERIFY OK: depth=1, C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=ca-certificate
Sat May 25 15:44:01 2019 VERIFY KU OK
Sat May 25 15:44:01 2019 Validating certificate extended key usage
Sat May 25 15:44:01 2019 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Sat May 25 15:44:01 2019 VERIFY EKU OK
Sat May 25 15:44:01 2019 VERIFY X509NAME OK: C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=glass-exercise-erode
Sat May 25 15:44:01 2019 VERIFY OK: depth=0, C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=glass-exercise-erode
```
**Sat May 25 15:44:02 2019 Connection reset, restarting [0]**
```
Sat May 25 15:44:02 2019 SIGUSR1[soft,connection-reset] received, process restarting
Sat May 25 15:44:02 2019 MANAGEMENT: >STATE:1558788242,RECONNECTING,connection-reset,,,,,
Sat May 25 15:44:02 2019 Restart pause, 5 second(s)
Sat May 25 15:44:07 2019 TCP/UDP: Preserving recently used remote address: [AF_INET]104.248.138.30:636
Sat May 25 15:44:07 2019 Socket Buffers: R=[8192->8192] S=[8192->8192]
Sat May 25 15:44:07 2019 Attempting to establish TCP connection with [AF_INET]104.248.138.30:636 [nonblock]
Sat May 25 15:44:07 2019 MANAGEMENT: >STATE:1558788247,TCP_CONNECT,,,,,,
Sat May 25 15:44:14 2019 TCP connection established with [AF_INET]104.248.138.30:636
Sat May 25 15:44:14 2019 TCP_CLIENT link local: (not bound)
Sat May 25 15:44:14 2019 TCP_CLIENT link remote: [AF_INET]104.248.138.30:636
Sat May 25 15:44:14 2019 MANAGEMENT: >STATE:1558788254,WAIT,,,,,,
Sat May 25 15:44:14 2019 MANAGEMENT: >STATE:1558788254,AUTH,,,,,,
Sat May 25 15:44:14 2019 TLS: Initial packet from [AF_INET]104.248.138.30:636, sid=7a5a6c2b f20ba6b5
Sat May 25 15:44:14 2019 VERIFY OK: depth=1, C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=ca-certificate
Sat May 25 15:44:14 2019 VERIFY KU OK
Sat May 25 15:44:14 2019 Validating certificate extended key usage
Sat May 25 15:44:14 2019 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Sat May 25 15:44:14 2019 VERIFY EKU OK
Sat May 25 15:44:14 2019 VERIFY X509NAME OK: C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=glass-exercise-erode
Sat May 25 15:44:14 2019 VERIFY OK: depth=0, C=US, ST=California, L=Beverly Hills, O=ACME CORPORATION, OU=Anvil Department, CN=glass-exercise-erode
Sat May 25 15:44:14 2019 Connection reset, restarting [0]
Sat May 25 15:44:14 2019 SIGUSR1[soft,connection-reset] received, process restarting
Sat May 25 15:44:14 2019 MANAGEMENT: >STATE:1558788254,RECONNECTING,connection-reset,,,,,
```
**Sat May 25 15:44:14 2019 Restart pause, 5 second(s)**

### Ansible Information

* Ansible version: 2.6.5
* Ansible system: Linux
* Host OS: Ubuntu
* Host OS version:  16.04
* Python interpreter: /usr/bin/python
* Python version: 2.7.12

### Streisand Information

* Streisand Git revision: c91292a71e6dc143141187eb79420aa14f8de2b0
* Streisand Git clone has untracked changes: no
* Genesis role: localhost
* Custom SSH key: True

### Enabled Roles

* Shadowsocks enabled:  True
* Wireguard enabled: True
* OpenVPN enabled: True
* stunnel enabled: True
* Tor enabled: False
* Openconnect enabled: True
* TinyProxy enabled: True
* SSH forward user enabled: True
* Configured number of VPN clients: 5

### Additional Details:
**_Log output from Ansible or other relevant services (link to Gist for longer output)_:**
**_Target Cloud Provider:_** 
Digital Ocean

**_Operating System of target host:_**
Ubuntu 16.04.6 LTS

**_Operating System of client:_**
Windows 7

**Version of Ansible, using ```ansible --version :```**
```
ansible 2.6.5
  config file = /root/streisand/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/streisand/venv/local/lib/python2.7/site-packages/ansible
  executable location = /root/streisand/venv/bin/ansible
  python version = 2.7.12 (default, Nov 12 2018, 14:36:49) [GCC 5.4.0 20160609]
```
**git rev-parse HEAD** 
c91292a71e6dc143141187eb79420aa14f8de2b0

### More installation errors
```
TASK [download-and-verify : Verify the OpenVPN Community download signatures were from the correct keys] *****************************

failed: [localhost] (item=None) => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}
failed: [localhost] (item=None) => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}
fatal: [localhost]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

TASK [openvpn : One or more of the VPN clients could not be mirrored. Please file a bug report on GitHub so that the version number, checksum, or download location can be updated. Setup will now continue.] ***
Pausing for 20 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)

failed: [localhost] (item={u'sig': u'stunnel-latest-installer.exe.asc', u'file': u'stunnel-latest-installer.exe'}) => {"changed": false, "dest": "/var/www/streisand/mirror/stunnel/stunnel-latest-installer.exe", "item": {"file": "stunnel-latest-installer.exe", "sig": "stunnel-latest-installer.exe.asc"}, "msg": "Request failed", "response": "HTTP Error 404: Not Found", "state": "absent", "status_code": 404, "url": "https://www.stunnel.org/stunnel-latest-installer.exe"}
changed: [localhost] => (item={u'sig': u'stunnel-latest.tar.gz.asc', u'file': u'stunnel-latest.tar.gz'})

TASK [stunnel : One or more of the VPN clients could not be mirrored. Please file a bug report on GitHub so that the version number, checksum, or download location can be updated. Setup will now continue.] ***
Pausing for 20 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)


TASK [streisand-gateway : Attempt to open the instructions on Linux (if applicable). Errors in this task are ignored because the `xdg-open` command is not always available.] ***
fatal: [localhost -> localhost]: FAILED! => {"changed": false, "cmd": "xdg-open ../generated-docs/streisand-frav0.html", "msg": "[Errno 2] No such file or directory", "rc": 2}
...ignoring

TASK [streisand-gateway : Open the instructions on macOS (if applicable)] ************************************************************

skipping: [localhost]

PLAY RECAP ***************************************************************************************************************************

localhost                  : ok=349  changed=281  unreachable=0    failed=3
```

**Comment by julian45** [comment]
This appears to have been addressed [here](https://github.com/StreisandEffect/streisand/issues/1563#issuecomment-484270946); this is the same fix I applied when I had a similar issue

**Comment by rdbox** [comment]
> This appears to have been addressed [here](https://github.com/StreisandEffect/streisand/issues/1563#issuecomment-484270946); this is the same fix I applied when I had a similar issue

I tried this solution, the error did not help me.

**Comment by chrisb001** [comment]
Just realised that this might be the same issue I have with a new Ubuntu install: https://github.com/StreisandEffect/streisand/issues/1582. 

I also tried the approach LimitNPROC but this didn't help for me either.

**Comment by rdbox** [comment]
> I also tried the approach LimitNPROC but this didn't help for me either.

then there is no solution waiting for the main developers?

**Comment by chrisb001** [comment]
hi rdbox

I did get the LimitNPROC fix to work in the end. I couldn't get this to work at first until I read lunarthegrey's post carefully. You have to comment it out in 3 places:
/lib/systemd/system/openvpn@.service
/etc/systemd/system/openvpn@server-udp.service
/etc/systemd/system/openvpn@server.service

See: #1540 (comment)

**Comment by rdbox** [comment]
@chrisb001 Hi.

thank you very much.
Best advice, everything works great.
My mood today is like that of the happiest panda.

**Comment by rdbox** [comment]
Only my question is why the streisand developers make such mistakes. After all, in the old versions everything was fine

**Comment by lunarthegrey** [comment]
Just need to get my PR merged and no one will have this issue on new deploys.. https://github.com/StreisandEffect/streisand/pull/1553

Unfortunately no maintainers are responding.

**Comment by CorbanR** [comment]
@lunarthegrey @nopdotcom i've approved and merged in #1553 with the limit increase. So far so good setting up new servers (openvpn is working like a champ). Ill give it a few more days, and then close this issue(assuming no one else complains)
